html/js/tree/Player.js

```diff
@@ -0,0 +1,65 @@
+GTE.TREE = (function (parentModule) {
+    "use strict";
+
+    /**
+    * Creates a new Player.
+    * @class
+    * @param {String} name Player's name.
+    */
+    function Player(id, colour) {
+        this.id = id;
+        if (this.id === 0) {
+            this.name = GTE.PLAYERS.DEFAULT_CHANCE_NAME;
+            this.defaultName = GTE.PLAYERS.DEFAULT_CHANCE_NAME;
+        } else {
+            this.name = "" + this.id;
+            this.defaultName = "" + this.id;
+        }
+        this.colour = colour;
+    }
+
+    /**
+    * ToString function
+    */
+    Player.prototype.toString = function () {
+        return "Player: " + this.name;
+    };
+
+    /**
+    * Changes player's name to a given one
+    * @param {String} newName New player's name
+    */
+    Player.prototype.changeName = function (newName) {
+        try {
+            if (newName !== "") {
+                this.name = newName;
+            } else {
+                throw "Empty name";
+            }
+        } catch (err) {
+            console.log("EXCEPTION: " + err);
+        }
+    };
+
+    Player.prototype.draw = function (x, y) {
+        var thisPlayer = this;
+        return new GTE.UI.Widgets.ContentEditable(x, y,
+                GTE.CONSTANTS.CONTENT_EDITABLE_GROW_TO_RIGHT,
+                thisPlayer.name)
+                .colour(thisPlayer.colour)
+                .onSave(function () {
+                    var text = this.getText().replace(/&nbsp;/gi,'').trim();
```

Comment:
Can you leave a small comment explaining this line? e.g. // Removes all blankspace